woocommerce/woocommerce-gateway-stripe
# PR: Remove Stripe Checkout (the legacy modal)

Fixes #785

#### Changes proposed in this Pull Request:

Based on our discussions with Stripe and SCA-related work, and as described in @allendav's [comment](https://github.com/woocommerce/woocommerce-gateway-stripe/issues/785#issuecomment-481456778), we will remove support for Stripe Checkout in 4.2.0

This PR removes all traces of Stripe Checkout, including:

1. The main part of the code, located in `stripe.js` and `WC_Gateway_Stripe`, which handles the modal
2. All necessary settings, their docs (.md files), and the admin, which controls the visibility of those fields
3. Translations
4. Screenshot from the WP.org readme
5. Admin tests
6. Support for charges (instead of intents) in `WC_Gateway_Stripe`. This should not affect any extensions, since `WC_Gateway_Stripe::process_payment` is only responsible for one-time and initial payments.

#### Testing instructions

1. Navigate to the gateway's settings and ensure that there are no Stripe Checkout-related settings.
2. Do a few smoke tests with test payments, and make sure nothing blows up.
